backend/compact-connect/lambdas/purchases/handlers/privileges.py

```diff
@@ -42,3 +61,133 @@ def get_purchase_privilege_options(event: dict, context: LambdaContext):  # noqa
     options_response['items'] = serlialized_options
 
     return options_response
+
+
+def _find_best_license(all_licenses: list[dict]) -> dict | None:
+    if len(all_licenses) == 0:
+        return None
+
+    # Last issued active license, if there are any active licenses
+    latest_active_licenses = sorted(
+        [license_data for license_data in all_licenses if license_data['status'] == 'active'],
+        key=lambda x: x['dateOfIssuance'],
+        reverse=True,
+    )
+    if latest_active_licenses:
+        return latest_active_licenses[0]
+    # Last issued inactive license, otherwise
+    latest_licenses = sorted(all_licenses, key=lambda x: x['dateOfIssuance'], reverse=True)
+    return latest_licenses[0]
+
+
+@api_handler
+def post_purchase_privileges(event: dict, context: LambdaContext):  # noqa: ARG001 unused-argument
+    """
+    This endpoint allows a provider to purchase privileges.
+
+    The request body should include the selected jurisdiction privileges to purchase and billing information
+    in the following format:
+    {
+        "selectedJurisdictions": ["<jurisdiction postal code>"],
```

Comment:
should this be 'postal abbreviation'? (I assume these aren't zip codes like 95608).